vortexor/src/vortexor.rs

```diff
@@ -0,0 +1,216 @@
+        }
+    }
+}
+
+impl NotifyKeyUpdate for KeyUpdateNotifier {
+    fn update_key(&self, key: &Keypair) -> Result<(), Box<dyn std::error::Error>> {
+        let updaters = self.key_updaters.lock().unwrap();
+        for updater in updaters.iter() {
+            updater.update_key(key)?
+        }
+        Ok(())
+    }
+}
+
+impl Vortexor {
+    pub fn create_tpu_sockets(
+        bind_address: std::net::IpAddr,
+        dynamic_port_range: (u16, u16),
+        num_quic_endpoints: u64,
+    ) -> TpuSockets {
+        let quic_config = SocketConfig { reuseport: true };
+
+        let (_, tpu_quic) =
+            bind_in_range_with_config(bind_address, dynamic_port_range, quic_config.clone())
+                .expect("expected bind to succeed");
+
+        let tpu_quic_port = tpu_quic.local_addr().unwrap().port();
+        let tpu_quic = bind_more_with_config(
+            tpu_quic,
+            num_quic_endpoints.try_into().unwrap(),
+            quic_config.clone(),
+        )
+        .unwrap();
+
+        let (_, tpu_quic_fwd) = bind_in_range_with_config(
+            bind_address,
+            (tpu_quic_port.saturating_add(1), dynamic_port_range.1),
+            quic_config.clone(),
+        )
+        .expect("expected bind to succeed");
+
+        let tpu_quic_fwd = bind_more_with_config(
+            tpu_quic_fwd,
+            num_quic_endpoints.try_into().unwrap(),
+            quic_config,
+        )
+        .unwrap();
+
+        TpuSockets {
+            tpu_quic,
+            tpu_quic_fwd,
+        }
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    pub fn create_vortexor(
+        tpu_sockets: TpuSockets,
+        staked_nodes: Arc<RwLock<StakedNodes>>,
+        tpu_sender: Sender<PacketBatch>,
+        tpu_fwd_sender: Sender<PacketBatch>,
+        max_connections_per_peer: u64,
+        max_tpu_staked_connections: u64,
+        max_tpu_unstaked_connections: u64,
+        max_streams_per_ms: u64,
+        max_connections_per_ipaddr_per_min: u64,
+        tpu_coalesce: Duration,
+        identity_keypair: &Keypair,
+        exit: Arc<AtomicBool>,
+    ) -> Self {
+        let config = TpuStreamerConfig {
+            tpu_thread_name: "solQuicTpu",
+            tpu_metrics_name: "quic_streamer_tpu",
+            tpu_fwd_thread_name: "solQuicTpuFwd",
+            tpu_fwd_metrics_name: "quic_streamer_tpu_forwards",
+            quic_server_params: QuicServerParams {
+                max_connections_per_peer: max_connections_per_peer.try_into().unwrap(),
+                max_staked_connections: max_tpu_staked_connections.try_into().unwrap(),
+                max_unstaked_connections: max_tpu_unstaked_connections.try_into().unwrap(),
+                max_streams_per_ms,
+                max_connections_per_ipaddr_per_min,
+                wait_for_chunk_timeout: DEFAULT_WAIT_FOR_CHUNK_TIMEOUT,
+                coalesce: tpu_coalesce,
+            },
+        };
+
+        Vortexor::new(
```

Comment:
Why not merge the `new` with `create_vortexor`? We are creating the config object in this function which gets unraveled in `new`. It'll save on that.

Reply:
Will fix